state/protocol/badger/validity.go

```diff
@@ -111,3 +117,161 @@ func validCommit(commit *flow.EpochCommit, setup *flow.EpochSetup) error {
 
 	return nil
 }
+
+type receiptValidator struct {
+	state    protocol.State
+	index    storage.Index
+	results  storage.ExecutionResults
+	verifier module.Verifier
+}
+
+func NewReceiptValidator(state protocol.State, index storage.Index, results storage.ExecutionResults) protocol.ReceiptValidator {
+	rv := &receiptValidator{
+		state:    state,
+		index:    index,
+		results:  results,
+		verifier: signature.NewAggregationVerifier(encoding.ExecutionReceiptTag),
```

Comment:
Echoing same sentiment as in my [earlier comment](https://github.com/onflow/flow-go/pull/180#discussion_r530731295):
* for testability the implementation of `module.Verifier` should be injected into the `receiptValidator`

Regarding the specific implementation type that you need to use: 
* Check the data structure, whose signature you want to verify: https://github.com/onflow/flow-go/blob/d9d7f00ff2e9304f6c9e19088161ba40a1575123/model/flow/execution_receipt.go#L28-L33